sound/soc/sof/topology.c

```diff
@@ -2335,6 +2345,15 @@ static int sof_dai_load(struct snd_soc_component *scomp, int index,
 	if (!spcm->pcm.capture)
 		return ret;
 
+	ret = sof_parse_tokens(scomp, &spcm->stream[stream], stream_tokens,
+			       ARRAY_SIZE(stream_tokens), private->array,
+			       le32_to_cpu(private->size));
+	if (ret) {
+		dev_err(sdev->dev, "error: parse capture stream tokens failed %d\n",
+			le32_to_cpu(private->size));
+		return ret;
+	}
+
```

Comment:
@keyonjie I feel like this logic to call sof_parse_tokens() on stream_tokens is flawed. Assume the stream supports both playback and capture, then in the first sof_parse_tokens(), you will match both the PLAYBACK And CAPTURE tokens and wont you overwrite the PLAYBACK token value with the CAPTURE value?

Reply:
@ranj063 thanks for spotting this, I thought each sof_parse_tokens() invoking will parse a specific token only.

You are right, the sof_parse_tokens() calling will parse both tokens for us, so we need define different offset in the array, let me fix it and get back.